### PR TITLE
Add support for Oregon, USA

### PR DESCRIPTION
**Pull request by craftyguy**

### CODE REVIEW

**Comment by grote** [comment]
Hey, thanks a lot for your contribution! :)

Would you mind making a separate PR for the OregonProvider against public-transport-enabler? Then, the CI will also stop to fail. I don't want any more non-upstream providers in Transportr and eventually want to migrate the last ones to upstream as well.

**Comment by craftyguy** [comment]
Just to make sure I understand correctly, you want me to close this and then make a PR of this change to here: https://github.com/schildbach/public-transport-enabler/tree/master/enabler/src/de/schildbach/pte ?

**Comment by grote** [comment]
Let me clarify: You can leave this PR open, but remove `OregonProvider.java` from it (force push into your branch) and open a new PR against the PTE like this one here: schildbach/public-transport-enabler#76
As soon as it gets merged into PTE, I can merge this one.

Please not that inclusion in PTE might require some live tests (which are a good idea anyway). You can find an example in the PR I linked above. The `ParisProvider` has even more examples.

**Comment by craftyguy** [comment]
got it, thanks. I'll submit a PR for OregonProvider.java right now :)

I'm not quite sure where to find the stop_area (OML:SA:CTP3102842 in your exampel), etc used in the tests.. Do you have any pointers? I'm not all that familiar with Navitia, I just really want to use Transportr in Oregon :)

**Comment by grote** [comment]
> I'm not quite sure where to find the stop_area (OML:SA:CTP3102842 in your exampel), etc used in the tests.. Do you have any pointers?

This stop is specific to Quebec. I suggest you start with one `suggestLocations()` test. This will give you some locations in the output. If you use the command line, you might need to add `--info` for the test to produce output.

> I'm not all that familiar with Navitia, I just really want to use Transportr in Oregon :)

No problem, you will be able to use it soon! :)

File: res/values/strings.xml [hunk 1]
```diff
@@ -278,6 +278,7 @@ there are any).</string>
     <string name="np_desc_it_networks" translatable="false">ATM, GTT, AMAT, ACTV</string>
     <string name="np_desc_ontario">Ottawa, Toronto</string>
     <string name="np_desc_quebec">Deux-Montagnes, Laval, L\'Assomption, Outaouais, Sud-Ouest, Quebec, Haut-Saint-Laurent, Lanaudière, La Presqu\'Île, Laurentides, Montreal, Les Moulins, Vallée du Richelieu, Chambly-Richelieu-Carignan, Roussillon, Sorel-Varennes, Le Richelain, Sherbrooke, Sainte-Julie</string>
+    <string name="np_desc_oregon">Oregon, USA</string>
```
Comment (grote) [inline res/values/strings.xml]: I suggest you use that as a name (`setName()`) and use something for the description that somehow describes what is covered and what is not, available from here: https://navitia.opendatasoft.com/explore/dataset/us-or/table/?sort=type_file

**Comment by grote** [comment]
A vector drawable that represents the area or the transport companies operating in it would be nice as well.

**Comment by craftyguy** [comment]
Shoot, just saw your note about adding a SVG. I'll get that in and do yet another amend :)

File: src/de/grobox/liberario/TransportNetworks.java [hunk 2]
```diff
@@ -406,6 +406,12 @@ public TransportNetworks(Context context) {
 				         .setStatus(TransportNetwork.Status.BETA)
 		);
 
+		list.add(new TransportNetwork(context, NetworkId.OREGON)
+                                .setName(getString(R.string.np_name_oregon))
```
Comment (grote) [inline src/de/grobox/liberario/TransportNetworks.java]: Style Nitpick: this line has spaces, but should have tabs. You can see it is misaligned with the following lines.

**Comment by craftyguy** [comment]
I'm not completely sure why this is failing while the build succeeds for me on my system..

**Comment by grote** [comment]
That is because locally, you have the changes to the PTE already, while they are not included here. Let's get your PR merged in PTE. Still no news from Navitia? Maybe try again to get API access.

**Comment by craftyguy** [comment]
I used a different email and was able to finally register. I have a token, modified "secrets.properties" in the enabler to include my token, but when I run the test i get "could not find secredproperty file secrets.properties" even though that file is clearly in the class path.. Any suggestions?

**Comment by grote** [comment]
I have the same problem with Android Studio, need to [run the tests on the command line](https://github.com/schildbach/public-transport-enabler/blob/master/enabler/README.md).

**Comment by craftyguy** [comment]
Thank you for all of the help :)

I have submitted a test for the provider in my PR to PTE!

**Comment by grote** [comment]
Great! In the meantime, do you think it would be possible to turn the Oregon logo into a vector graphic?

**Comment by craftyguy** [comment]
Sorry for the newbie question, but where do you put SVGs in this project?

**Comment by grote** [comment]
SVG are placed in `/art/networks/`, but just as a reference. Android needs a special XML format. You can use a converter like [this one](https://inloop.github.io/svg2android/) and then place the result in `/res/drawable/`. Check the other network drawables for examples.

**Comment by craftyguy** [comment]
Is the SVG absolutely necessary before this is accepted for merging? I'm not sure when I'll have time to sit down and figure it out.. and it seems like having this provider in would be useful (at least for me, but hopefully for others!)

**Comment by grote** [comment]
No, we can merge it before, but we are waiting for the upstream PR to be merged anyway...

**Comment by craftyguy** [comment]
So...is there anything I need to do to help this get merged?

**Comment by grote** [comment]
Well this PR has a merge conflict, but it only makes sense to update it once the upstream PTE support as been merged and there you can only keep pestering @schilbach I guess ;)

**Comment by grote** [comment]
@craftyguy I am closing this for now, because support for Oregon has still not been merged into PTE and this PR is now outdated. I don't know why this is taking so long :(

Feel free to push for PTE inclusion and then reopen this PR.

**Comment by craftyguy** [comment]
booo :(

**Comment by craftyguy** [comment]
@grote

PTE merged my PR! Can this PR be reopened?

**Comment by grote** [comment]
Of course, but it would still need to be rebased on latest master, because it does not apply cleanly anymore.

**Comment by craftyguy** [comment]
Thank you, I'll take a look at rebasing it as soon as possible!

**Comment by craftyguy** [comment]
I've tried to rebase, but github thinks my modifications are "merge conflicts." Not sure what could/did go wrong..

**Comment by patkan** [comment]
I am not sure what you did, but you did not rebase to master ;-)
Have you used something like the following?
```
git fetch upstream
git checkout add_oregon_provider
git rebase upstream/master
```
If you are not sure you can just do it on a new branch. `git checkout -b new-branch` creates a new one branching your current.

**Comment by grote** [comment]
Thanks @patkan for helping @craftyguy with his PR. That is very much appreciated, so I don't need to do it and can focus on other work :)

> Edit: I haven't tested your commit. As I look at it now I think you forgot to include the new version of 
pte.

I'll pull in the new version in master, so that will be no issue here.

**Comment by craftyguy** [comment]
@patkan thank you so much for the help. I do this often enough that I always forget the appropriate order of operation. I went ahead and took another stab at it before looking at your repo just to see if I had learned anything, and it seems like it might finally be straight :)

**Comment by grote** [comment]
Thanks for your patience @craftyguy! I'll merge this now and then it will be part of the next release.

**Comment by craftyguy** [comment]
@grote Great! Sorry for the long and drawn out PR... I might try to do another one just for fun :P

**Comment by grote** [comment]
No problem! I am quite happy about your PR and would love to see a new one. If you want to get inspired, there's a list of smaller jobs here: https://github.com/grote/Transportr/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+job%22

**Pull request merged**